experimenter/experimenter.py

```diff
@@ -181,42 +184,41 @@ def _generate_standard_pipeline(self, preprocessor: PrimitiveBase, classifier: P
         # Creating Pipeline
         pipeline_description = Pipeline(context=Context.TESTING)
 
-        step_counter = self._add_initial_steps(pipeline_description, step_counter)
+        step_counter, attributes_step_i, target_step_i = self._add_initial_steps(pipeline_description, step_counter)
         preprocessor_used = False
 
-        # Step 4: Preprocessor
+        # Preprocessor Step
         if preprocessor:
-            step_4 = PrimitiveStep(primitive_description=preprocessor.metadata.query())
+            preprocessor_step = PrimitiveStep(primitive_description=preprocessor.metadata.query())
             for arg in self._get_required_args(preprocessor):
                 if arg == "outputs":
-                    data_ref = 'steps.3.produce'
+                    data_ref = f'steps.{target_step_i}.produce'
                 else:
-                    data_ref = 'steps.{}.produce'.format(step_counter - 2)
+                    data_ref = f'steps.{attributes_step_i}.produce'
                     preprocessor_used = True
 
-                step_4.add_argument(name=arg, argument_type=ArgumentType.CONTAINER,
+                preprocessor_step.add_argument(name=arg, argument_type=ArgumentType.CONTAINER,
                                     data_reference=data_ref)
-            step_4.add_hyperparameter(name='use_semantic_types', argument_type=ArgumentType.VALUE, data=True)
-            step_4.add_hyperparameter(name='return_result', argument_type=ArgumentType.VALUE, data="replace")
-            step_4.add_output('produce')
-            pipeline_description.add_step(step_4)
+            preprocessor_step.add_hyperparameter(name='use_semantic_types', argument_type=ArgumentType.VALUE, data=True)
+            preprocessor_step.add_hyperparameter(name='return_result', argument_type=ArgumentType.VALUE, data="replace")
+            preprocessor_step.add_output('produce')
+            pipeline_description.add_step(preprocessor_step)
             step_counter += 1
 
-        # Step 5: Classifier
-        step_5 = PrimitiveStep(primitive_description=classifier.metadata.query())
-        for index, arg in enumerate(self._get_required_args(classifier)):
+        # Classifier Step
+        classifier_step = PrimitiveStep(primitive_description=classifier.metadata.query())
+        for arg in self._get_required_args(classifier):
             if arg == "outputs":
```

Comment:
Oh I see we didn't change that.  Probably ignore my other comments.

Reply:
Please resolve the comments you want us to ignore, if you haven't done so.